### ISSUE TITLE
[bigquery-io] - Add ability to read data serialized in Arrow format

### ISSUE THREAD

**Issue by vlasenkoalexey** [tensorflow/io#891]
Deserializing Arrow should be faster than Avro

**Comment by yongtang** [comment]
Also /cc @BryanCutler for arrow-related.

**Comment by vlasenkoalexey** [comment]
@BryanCutler, I looked at Arrow code, as far as I understand ArrowUtil::AssignTensor takes an array of Arrow serialized records and converts them to Tensor. Is it correct?
Also would you mind if I move arrow_util into a separate library so OPs don't depend on each other?

**Comment by yongtang** [comment]
+1 on separate library. /cc @BryanCutler

**Comment by BryanCutler** [comment]
> ArrowUtil::AssignTensor takes an array of Arrow serialized records and converts them to Tensor. Is it correct?
@vlasenkoalexey  yes that is correct. The tensor must also be pre-allocated large enough to fit the Arrow array data. Moving arrow_util into a separate library sounds great, thanks!

**Comment by vlasenkoalexey** [comment]
@BryanCutler I made prototype work, but noticed that Arrow doesn't support string to TF tensor translation:
https://github.com/apache/arrow/blob/master/cpp/src/arrow/adapters/tensorflow/convert.h
And at the same time I see that Arrow can handle strings just fine.
Do you happen to know why such limitation exists?
Would you mind if extend AssignTensor and CheckArrayType to support strings?

**Comment by BryanCutler** [comment]
Hi @vlasenkoalexey , the limitation was just due to not having the time to add it in. If you could add this, that would be great, thanks!

**Comment by vlasenkoalexey** [comment]
SG, I'll give it a shot.
